test/testlib.sh

```diff
@@ -45,6 +45,21 @@ DIR_BAZ=$(realdir "$PWD/$net3")
 
 # Register helper functions
 
+if [ "$(uname -s)" = SunOS ]; then
+  gnu=/usr/gnu/bin
+  grep="$gnu/grep"
+
+  grep() { $gnu/grep "$@"; }
+  tail() { $gnu/tail "$@"; }
+
+  if ! tail /dev/null || ! echo '' | grep ''; then
+    echo >&2 'Sorry, native Solaris tools are not supported. Please install GNU Coreutils.'
+    exit $EXIT_SKIP_TEST
+  fi
+else
+  grep='grep'
+fi
+
```

Comment:
As @eli-schwartz mentioned in #357, meson already depends on Python, so that makes me wonder if we should not just rewrite all the test scripts in Python. But that's something for another pull request.

Reply:
In theory there is also https://sr.ht/~lattis/muon /https://muon.build which is a c99 rewrite of Meson with a minimal bootstrap path and suitable for addressing https://mesonbuild.com/FAQ.html#but-i-really-want-a-version-of-meson-that-doesnt-use-python

Not everyone cares about that, of course. Practically speaking, depending on python for the build machine isn't so unreasonable. And muon doesn't care if you use scripts which depend on python, it will still work (at least if you don't use muon's current major weak point which is modules support). But you will of course no longer be able to depend on muon allowing you to knock python out of the build requirements.

I'd consider that benefit to be mainly of interest to people doing systems bootstrapping especially in cases where you are operating a self-hosting Linux distro where base features like the init system or C library use Meson to build. That's actually a major consideration for Alpine Linux and OpenRC.

I don't know if you care. Probably not, as tinc isn't really a distro bootstrap package.

Reply:
The idea already came up when we were improving tests last year, but was rejected because Python wasn't required back then.

Unix utilities are very similar… from a 10 000-foot view. When you start doing anything complicated, you run into lots of smalls differences.

So tests already have "non-native" dependencies because I couldn't make them work reliably everywhere: procps, coreutils, a particular flavor of netcat.

I think I'd rather replace all that with Python.

Reply:
Yeah, making things work for the least common denominator of "POSIX specifies this utility with this behavior" can be quite hard to build complex actions out of.

The only other alternative is probably to write the tests in C.